### DISCUSSION TITLE
Should inappropriate vocabulary be removed?

### DISCUSSION THREAD

**Discussion by rcfox** [faker-js/faker#1631] [Ideas]
To start off: I'm not a prude. I don't care about swearing and "dirty" language. However, I noticed the word "sodomise" show up in a test this morning, and that caught my attention. If that showed up in a demo with product management/executives/clients, that could potentially be pretty embarrassing.

At a glance, there are at least a [handful](https://github.com/faker-js/faker/blob/next/src/locales/en/word/verb.ts#L2261) [of](https://github.com/faker-js/faker/blob/next/src/locales/en/word/noun.ts#L4163) [questionable](https://github.com/faker-js/faker/blob/next/src/locales/en/word/verb.ts#L4805) [words](https://github.com/faker-js/faker/blob/next/src/locales/en/word/noun.ts#L5529) [mixed](https://github.com/faker-js/faker/blob/next/src/locales/en/word/noun.ts#L4495) [in](https://github.com/faker-js/faker/blob/next/src/locales/en/word/verb.ts#L3161).

(The library also includes [male reproductive anatomy](https://github.com/faker-js/faker/blob/next/src/locales/en/word/noun.ts#L4403) but fails to include female reproductive anatomy. Yet another instance of lack of female representation in the industry! ;) )

Again, I'm not trying to suck the fun out of the project, but it seems like Faker is targeted at a professional audience who might not appreciate these words cropping up, even if only used internally. These words would also seem to be deemed unacceptable by the [code of conduct](https://github.com/faker-js/faker/blob/6bac476695c146a794b5337785979a7395d7dc89/CODE_OF_CONDUCT.md)

Alternatively, perhaps a new `en_clean` locale could be created?

**Comment by ST-DDT** [comment]
These words should be removed.
We already removed some of them, but we havent checked every single word in every language (~200k entries). We mostly prevent new potentially offending/unproffessional words for new locale data.

Are you willing to create a PR that removes these values?

**Reply by rcfox** [reply]
Yeah, I can give the English lists a look over tonight or tomorrow. I'm not fluent in any other languages.

Are there guidelines on allowable words? Obviously sexual and vulgar terms are out. What about words like "slave"/"slavery" that have historical baggage or other negative connotations? I remember the common use of "master" in git repositories was a subject of some debate a few years ago within the tech community.

In my opinion, the English language is large enough that any individual word is unlikely to be missed, and it's better to just avoid headaches that could crop up from their inclusion.

**Reply by Shinigami92** [reply]
For me, words like `slavery` or `blacklist` are more bad to me as `masturbate` 🤷
So yeah, please Filter words like these out.

**Reply by rcfox** [reply]
I've put up a preliminary PR: https://github.com/faker-js/faker/pull/1633/files

The words I've been removing have for the most part fallen into these categories:

* Sexuality (including garments that imply sexual anatomy)
* Drugs/alcohol
* Serious disease
* Insults
* Ableism
* Curse words
* Racism
* Over-religiousness
* Generally distasteful ("beheading", etc.)
* Colours as nouns. (I first encountered "black" as a noun, which seemed somewhat racist, but then ran into "blue" and "brown", so I figured they should go too.)

I'm stopping here for tonight because my eyes need a rest, and I thought it'd be good to get some feedback on whether you think I'm filtering too much. I think for almost all of the words that I've removed, it would be difficult/embarrassing to explain to my boss why they showed up in my project.

Thanks.

**Comment by Shinigami92** [comment]
For the short term, I things yes: the best solution would be to just clean them out for now.
But in the long term, maybe somehow an explizit extension might be needed :thinking: 
But this would require extending loaded locales in a UX/DX friendly way, and currently I don't know how we could do that.

**Reply by ST-DDT** [reply]
> But in the long term, maybe somehow an explizit extension might be needed 🤔

For what purpose?

**Reply by Shinigami92** [reply]
Sex toy web shop, I don't know :shrug:

**Reply by ST-DDT** [reply]
That sounds more like a new category/entry (commerce/product) than a new locale. However, none that I would like to have.
"Specialized" shops should use their own product catalog or our generic one.
